contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -112,6 +112,7 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
                 newDefaultAdmin == address(0) && _isScheduleSet(schedule) && _hasSchedulePassed(schedule),
                 "AccessControl: only can renounce in two delayed steps"
             );
+            delete _pendingDefaultAdminSchedule;
```

Comment:
Small comment:

If a dev calls `_revokeRole` internally, the schedule is not reset. This means that if someone does a `beginTransferAdmin` and then a `_revokeRole`, the contract appears to be administered by `address(0)`, but the pending owner can still claim the ownership.

IMO this delete should go (in addition to `delete _pendingDefaultAdmin`) in `_revokeRole`.

```solidity
    function _revokeRole(bytes32 role, address account) internal virtual override {
        if (role == DEFAULT_ADMIN_ROLE && account == _currentDefaultAdmin) {
            delete _currentDefaultAdmin;
        }
        super._revokeRole(role, account);
    }
```

Reply:
I tested with 
```solidity
    function _revokeRole(bytes32 role, address account) internal virtual override {
        if (role == DEFAULT_ADMIN_ROLE && account == _currentDefaultAdmin) {
            delete _currentDefaultAdmin;
            delete _pendingDefaultAdmin;
            delete _pendingDefaultAdminSchedule;
        }
        super._revokeRole(role, account);
    }
```
and the tests pass.

Reply:
(I'm increasingly worried by the number of corner case that this contract has, and that we are discovering quite late IMO)

Reply:
> For example, a developer might want to extend the contract in a way in which a multisig has the ability to disable a DEFAULT_ADMIN before a new one enters into effect. That's a valid use case for using _revokeRole while keeping the already scheduled one.

That is something I raised on slack as being a possible attack: 
- you initiate a default admin transfer
- you use the internal `_revoke`. This emits the revoke event, and set the current admin to 0
  - at the point users will think the contract has no admin, and will not get a new one
- the pending admin accepts, and "resurrect" the admin rights
  - this goes against the expectation (by the user) that the admin was removed from the system.
  
If someone reported that on Immunefi, would we say its the expected behavior. As a user wouldn't expect this behavior.

I personally like the fact that my proposal makes `_acceptDefaultAdminTransfer` "cleaner"
- we cache the pending admin
- we check the schedule
- we use revoke to clear old state
- we use grant to set the new one

Reseting the state in revoke is clearly opinionated. I think what it comes down to is whether we consider the 
```
beginDefaultAdminTransfer → _revoke → acceptDefaultAdminTransfer
```
workflow being dangerous or not.

Reply:
Yeah I also don't see a reason why `_revokeRole` should reset the schedule.

> if someone does a `beginTransferAdmin` and then a `_revokeRole`, the contract appears to be administered by `address(0)`, but the pending owner can still claim the ownership.

This is an interesting observation but I don't think it's specific to this situation with revoke. In general, anyone that wants to check the permissions should always also check if there's a pending/scheduled admin. For example, the same strategy could be used to mislead someone into thinking that account A is the admin when in fact account B is a pending admin that can claim at any moment.

Reply:
> I agree with this statement. I'd be more in favor of an internal _renounceAdmin that does both the _revoke and the schedule reset, but _revoke resetting the schedule on its own sounds to me like a misleading assumption

Fair!